packages/ui/vue/vue-template/src/components/ResultLink.vue

```diff
@@ -0,0 +1,57 @@
+<template>
+  <a
+    @click="onSelect"
+    @contextmenu="onSelect"
+    @mousedown="onSelect"
+    @mouseup="onSelect"
+    @touchstart="beginDelayedSelect"
+    @touchend="cancelPendingSelect"
+    :href="filteredUri"
+    >{{ result.title }}</a
+  >
+</template>
+
+<script lang="ts">
+import { computed, inject } from "vue";
+import {
+  buildInteractiveResult,
+  type Result,
+  type SearchEngine,
+} from "@coveo/headless";
+import { HeadlessInjectionKey } from "@/headlessKey";
+
+let engine: SearchEngine;
+export default {
+  name: "ResultLink",
+  props: ["result"],
```

Comment:
Is this what causing the types for `Result` to be unknown/any ?

Reply:
All props are `type|undefined` because you have no way to know if the component user set the prop